woocommerce/google-listings-and-ads
# Free Listings + Paid Ads: Add the paid ads previews to the boost product listings section

### Changes proposed in this Pull Request:

This PR implements the ad previews part of **📌 Boost product listings section** in #1610.

- Dynamically compositing the ad previews with product and shop data.

- [x] 💡Please note that there is a very small white border in the fallback image of product cover. It's not a CSS style problem and I will check with the design team later.
   - Fixed in https://github.com/woocommerce/google-listings-and-ads/pull/1650/commits/6fe6ba71e1b43c78b8519ff2fc5cdd5f3ca40cc4.

### Screenshots:

#### 📷 Text ellipsis

![2022-08-24 16 04 00](https://user-images.githubusercontent.com/17420811/186405910-6df3bd4b-2c04-4a97-8435-fc2e9baafbf0.png)

#### 📷 Simulation of using product cover from merchant's DB

![2022-08-24 17 11 07](https://user-images.githubusercontent.com/17420811/186405922-b7f04a93-7030-4f63-a214-201530df5670.png)

#### 📷 Simulation of using shop logo from merchant's website

![2022-08-24 17 11 39](https://user-images.githubusercontent.com/17420811/186405936-c5015d3a-5a32-4d4f-825f-f1ae61b58744.png)

#### 📹 All ad previews and blur animation

https://user-images.githubusercontent.com/17420811/186405954-5cf6471b-7889-4fcc-9c69-88c8d2d14037.mp4

### Detailed test instructions:

1. Go to step 4 of the onboarding flow.
1. Check if the visual result of the ad previews in the "Boost product listings with paid ads" section is close to the design in Figma.
1. Change [these values](https://github.com/woocommerce/google-listings-and-ads/blob/05779b932a21b71b8d7aa41321fe097b6f122bdb/js/src/components/paid-ads/campaign-preview/campaign-preview.js#L29-L48) locally to see if the layouts of ad previews are still rendered properly.
